packages/raam/src/index.ts

```diff
@@ -1,8 +1,267 @@
-export * from "./inline";
-export * from "./stack";
-export * from "./wrap";
+import * as CSSType from "csstype";
+
+// @TODO Ensure RaamCSS is valid with other CSS libraries. Right now it throws
+// incompatibility errors
+type RaamCSSProperties = CSSType.PropertiesFallback;
+type RaamCSS =
+  | RaamCSSProperties
+  | { [P in CSSType.SimplePseudos | string]: RaamCSSProperties };
+
+type RaamStyleOption<K extends keyof RaamCSS, T = void> =
+  | RaamCSS[K]
+  | T
+  | { [mediaQuery: string]: RaamCSS[K] | T };
+
+type RaamStyleOptions<K extends keyof RaamCSS, T = void> = {
+  [P in K]?: RaamStyleOption<K, T> | RaamStyleOption<K, T>[];
+};
+
+enum CSS_VARS {
+  FLEX_GAP_OFFSET = "--raam-fg-offset",
+  FLEX_GAP = "--raam-fg",
+  FLEX_GAP_TOP = "--raam-fg-t",
+  FLEX_GAP_LEFT = "--raam-fg-l",
+}
+
+// Utils
+// ==============================================
+
+// Polyfill for `.flat()`
+const flat = (arr: any[]) =>
+  arr.reduce(
+    (accumulator, currentValue) =>
+      accumulator.concat(
+        Array.isArray(currentValue) ? flat(currentValue) : currentValue
+      ),
+    []
+  );
+
+const reset = (as: unknown): RaamCSS => ({
+  // @TODO Possibly bin-off these 3 initial styles to avoid duplication
+  boxSizing: "border-box",
+  margin: 0,
+  minWidth: 0,
+  ...((as &&
+    typeof as === "string" &&
+    {
+      /**
+       * Override browser defaults.
+       */
+      ul: { paddingLeft: 0 },
+      /**
+       * Ensure any list-items are rendered without bullets.
+       * Adding a zero-width space in the content to prevent VoiceOver disable.
+       */
+      li: {
+        listStyleType: "none",
+        "&:before": {
+          position: "absolute",
+          content: '"\\200B"',
+        },
+      },
+    }[as]) ||
+    {}),
+});
+
+type RaamFlexGap = {
+  flexGap?: RaamStyleOption<"gap">;
+};
+type StyleOptionsToCSS = RaamStyleOptions<keyof RaamCSS> & RaamFlexGap;
+
+const styleOptionGetValue = (
+  // @TODO This probably shouldn't be a string type
+  property: keyof StyleOptionsToCSS | string,
+  originalValue
+) => {
+  const getValue =
+    (typeof originalValue === "object" && Object.values(originalValue)[0]) ||
+    originalValue;
+
+  if (property === "flexGap") {
+    return typeof getValue === "number" ? `${getValue}px` : getValue;
+  }
+
+  return getValue;
+};
+
+const styleOptionToCSS = (
+  // @TODO This probably shouldn't be a string type
+  property: keyof StyleOptionsToCSS | string,
+  originalValue
+): RaamCSS => {
+  const breakpoint =
+    typeof originalValue === "object" && Object.keys(originalValue)[0];
+
+  const value = styleOptionGetValue(property, originalValue);
+
+  const polyfill = {
+    /**
+     * Replace `flexGap` with custom properties.
+     * These are toggled off (via `initial`) by the appropriate flex property
+     * in the latter step.
+     */
+    flexGap: {
+      [CSS_VARS.FLEX_GAP_TOP]: value,
+      [CSS_VARS.FLEX_GAP_LEFT]: value,
+      [CSS_VARS.FLEX_GAP_OFFSET]: `calc(${value} * -0.5)`,
```

Comment:
Using `calc(var(--raam-fg) * -1)` (rendered css) here would make it a little easier to update the gap size size you'll only need to adjust `--raam-fg`

Reply:
Thanks for this! To be honest, I just went for this approach because I had a hunch there were perf concerns when it comes to using `calc` with aliased css custom properties - I can't find anything to back that up though. Maybe something to dig into before changing, just in case

Reply:
fwiw, the `calc()` function is calculated only once, see https://codepen.io/MartijnCuppens/pen/ZEOYxoY?editors=1100
The `--raam-fg` property is defined in the element itself, so I don think there won't be any significant performance issues.

Reply:
I'll dig into this some more, thanks for the suggestion and example